src/mumble/AudioOutput.cpp

```diff
@@ -431,23 +446,18 @@ bool AudioOutput::mix(void *outbuff, unsigned int frameCount) {
 
 	bool prioritySpeakerActive = false;
 
-	// Get the users that are currently talking (and are thus serving as an audio source)
+	// Detect whether priority speaker is active.
 	QMultiHash< const ClientUser *, AudioOutputBuffer * >::const_iterator it = qmOutputs.constBegin();
 	while (it != qmOutputs.constEnd()) {
-		AudioOutputBuffer *buffer = it.value();
-		if (!buffer->prepareSampleBuffer(frameCount)) {
-			qlDel.append(buffer);
-		} else {
-			qlMix.append(buffer);
-
-			const ClientUser *user = it.key();
-			if (user && user->bPrioritySpeaker) {
-				prioritySpeakerActive = true;
-			}
+		const ClientUser *user = it.key();
+		if (user && user->bPrioritySpeaker) {
```

Comment:
You lost the check here if ``buffer->prepareSampleBuffer`` returned true, which is only the case when the buffer is not done yet. Without digging deeper I am not sure, if this causes unintended side effects.

Reply:
Ah yes, that's true. However, it will not have an effect because empty buffers are discarded at the end of `mix()`. If a priority speaker's buffer is available but finished, it will affect only the next `frameCount` samples before it is removed again. It boils down to whether this is an acceptable error or not. Considering that buffer sizes are usually fractions of a second (< 0.1s), I'd say its negligible.

Reply:
It becomes a question of whether the buffer size of the JACK server (typically 1024 frames or less) represents a larger amount of time than the amount of error already present in detecting when the priority speaker has "stopped speaking." this includes network latency and, if the priority speaker is using voice activity instead of push to talk, whatever latency exists in that subsystem's output as well. 

Furthermore, 1024 frames at 48 kHz is just under a 48th of a second; Basically all natural pauses in conversational speech (even spaces between individual words) are more than that. This is why I doubt that it matters.

Reply:
If this is only about extending the lifetime of a priority speaker (which, when thinking about it again, it probably is), then yes I have to agree with you. I was only worried that this might jump at us in between a stream of continuous audio, but that doesn't make sense.

Reply:
Correct, it only extends the lifetime by `frameCount` frames. Glad I could make my perspective clear! :D